File: frontend/lib/domain/entities/course.dart
// frontend/lib/domain/entities/course.dart
import 'package:equatable/equatable.dart';

class Course extends Equatable {
  final int id;
  final String title;
  final String description;
  final String tier;
  final String? thumbnailUrl;
  final String? provider;
  final String? url;

  const Course({
    required this.id,
    required this.title,
    required this.description,
    required this.tier,
    this.thumbnailUrl,
    this.provider,
    this.url,
  });

  @override
  List<Object?> get props => [id, title, description, tier, thumbnailUrl, provider, url];
}

File: backend/app/application/dtos/course.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from .lesson import Lesson
from app.domain.value_objects.track import Track

class CourseBase(BaseModel):
    title: str
    description: str
    tier: Track
    thumbnail_url: Optional[str] = None

class CourseCreate(CourseBase):
    pass

class Course(CourseBase):
    id: int
    lessons: List["Lesson"] = []

    model_config = ConfigDict(from_attributes=True)

Course.model_rebuild()
